client/state/user-devices/reducer.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { keyBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	USER_DEVICES_REQUEST,
+	USER_DEVICES_REQUEST_SUCCESS,
+} from 'state/action-types';
+
+export const items = createReducer( {}, {
+	[ USER_DEVICES_REQUEST_SUCCESS ]: ( state, { devices } ) => ( { ...keyBy( devices, 'device_id' ) } )
```

Comment:
Ah I missed, Dennis's note earlier on this. `USER_DEVICES_ADD` also works for me.